tls/s2n_recv.c

```diff
@@ -51,7 +52,10 @@ int s2n_read_full_record(struct s2n_connection *conn, uint8_t * record_type, int
 
     /* Read the record until we at least have a header */
     while (s2n_stuffer_data_available(&conn->header_in) < S2N_TLS_RECORD_HEADER_LENGTH) {
-        r = s2n_stuffer_recv_from_fd(&conn->header_in, conn->readfd, S2N_TLS_RECORD_HEADER_LENGTH - s2n_stuffer_data_available(&conn->header_in));
+        int remaining = S2N_TLS_RECORD_HEADER_LENGTH - s2n_stuffer_data_available(&conn->header_in);
+
+        GUARD(s2n_socket_read_size(conn, remaining));
```

Comment:
nit: s2n_socket_read_size doesn't make it clear that the funciton is mutating the socket options. Maybe s2n_socket_set_read_size/s2n_socket_set_rcvlowat would be better?